### PR TITLE
Update JitOptimizerTodoAssessment.md

### PR DESCRIPTION
**Pull request by JosephTremoulet**
Add a few new issues and update a few based on recent developments.  Also, categorize the list.

### CODE REVIEW

File: Documentation/performance/JitOptimizerTodoAssessment.md [hunk 1]
```diff
@@ -99,36 +122,52 @@ benchmarking story set up to measure the effect of such changes when they go in,
...
+-------------
+
+### Value Number Conservativism
+
+We have some frustrating phase-ordering issues resulting from this, but the
+opt-repeat experiment indicated that they're not prevalent enough to merit
+pursuing changing this right now.  Also, using SSA def as the proxy for value
+number would require handling SSA renaming, so there's a big dependency chained
+to this.
+Maybe it's worth reconsidering the priority based on throughput?
```
Comment (russellhadley) [inline Documentation/performance/JitOptimizerTodoAssessment.md]: What's the throughput argument?  Do we have a rough t-shirt size based on one of our scenarios?

**Comment by russellhadley** [comment]
Looks good.  I had one question and this last comment.  It might be nice to have a list of items in flight so we can keep an accounting of what things we check off.  Doesn't need to be for everything but when we make a GH issue and start analyzing/working on it that might be the right point to track it.

File: Documentation/performance/JitOptimizerTodoAssessment.md (hunk 1 above)
Comment (JosephTremoulet) [inline Documentation/performance/JitOptimizerTodoAssessment.md]: The throughput argument is that it's wasteful having a side-table of expressions in a separate expression language, if we used an eager rewrite strategy the hash table could point back to the IR itself to record the expressions.  We would need to somehow record the information about which stores reach which loads that's currently also in the VN tables, though, so it's not like the VN memory is totally redundant.  No, I don't have numerical estimates on this.

**Comment by JosephTremoulet** [comment]
> It might be nice to have a list of items in flight so we can keep an accounting of what things we check off. Doesn't need to be for everything but when we make a GH issue and start analyzing/working on it that might be the right point to track it.

I'm not following what the benefit is of PRing that info to a document; if we're just disciplined about self-assigning issues we're looking into, then a query of assigned codegen perf issues would give you that list without extra overhead.  And if we update the issues as we make progress, a query of recently modified codegen perf issues works too.

**Comment by JosephTremoulet** [comment]
I pushed an update with a description of jit helper call kill set improvements that I somehow omitted from the original PR.

**Pull request merged**